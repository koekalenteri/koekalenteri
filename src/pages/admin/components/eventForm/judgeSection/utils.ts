import type { DeepPartial, EventClass, EventType, Judge, PublicJudge } from '../../../../../types'
import type { PartialEvent } from '../../EventForm'

import { isSameDay } from 'date-fns'

export type PartialPublicJudge = Partial<PublicJudge>
export type PartialPublicJudgeValue = PartialPublicJudge | PartialPublicJudge[]

export const filterJudges = (
  judges: Judge[],
  eventJudges: PublicJudge[],
  id: number | undefined,
  eventType?: EventType
) =>
  judges
    .filter((j) => !eventType?.official || j.eventTypes.includes(eventType.eventType))
    .filter((j) => j.id === id || !eventJudges.find((ej) => ej.id === j.id))

export const hasJudge = (c: DeepPartial<EventClass>, id?: number) =>
  Array.isArray(c.judge) ? c.judge.find((j) => j.id === id) : c.judge?.id === id

export const filterClassesByJudgeId = (classes?: EventClass[], id?: number) => classes?.filter((c) => hasJudge(c, id))

export const toArray = (j?: PartialPublicJudge): PartialPublicJudge[] => (j ? [j] : [])
export const makeArray = (j?: PartialPublicJudgeValue) => (Array.isArray(j) ? [...j] : toArray(j))
export const selectJudge = (j?: PartialPublicJudgeValue, judge?: PublicJudge): PartialPublicJudge[] => {
  const a = makeArray(j)
  if (judge) {
    const index = a.findIndex((cj) => cj.id === judge.id)
    if (index === -1) {
      a.push(judge)
    } else {
      a[index] = judge
    }
  }
  return a
}

export const removeJudge = (j?: PartialPublicJudgeValue, id?: number): PartialPublicJudge[] => {
  const a = makeArray(j)
  return a.filter((cj) => cj.id !== id)
}

export const updateJudge = (
  event: PartialEvent,
  id: number | undefined,
  judge: PublicJudge | undefined,
  values?: DeepPartial<EventClass>[]
) => {
  const isSelected = (c: DeepPartial<EventClass>) =>
    values?.find((v) => event && isSameDay(v.date ?? event.startDate, c.date ?? event.startDate) && v.class === c.class)
  return event.classes?.map((c) => ({
    ...c,
    judge: isSelected(c) ? selectJudge(c.judge, judge) : removeJudge(c.judge, id),
  }))
}
