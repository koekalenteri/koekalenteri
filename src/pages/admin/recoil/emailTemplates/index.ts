export * from './actions'
export * from './atoms'
export * from './atoms2'
export * from './effects'
export * from './selectors'
