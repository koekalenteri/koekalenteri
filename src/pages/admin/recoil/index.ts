export * from './emailTemplates'
export * from './events'
export * from './eventTypes'
export * from './officials'
export * from './organizers'
export * from './registrations'
export * from './user'
