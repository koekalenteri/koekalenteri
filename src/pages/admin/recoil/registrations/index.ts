export * from './atoms'
// 2nd file to avoid circular deps
export * from './atoms2'
export * from './selectors'
