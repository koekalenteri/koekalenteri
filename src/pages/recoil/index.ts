export * from './effects'
export * from './events'
export * from './judges'
export * from './registration'
export * from './user'
