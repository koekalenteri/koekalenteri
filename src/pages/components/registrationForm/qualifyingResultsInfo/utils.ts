import type { EventResultRequirement, EventResultRequirements, EventResultRequirementsByDate } from '../../../../rules'
import type { ManualTestResult, QualifyingResult } from '../../../../types'

import { nanoid } from 'nanoid'

import { unique } from '../../../../lib/utils'
import { objectContains } from '../validation'

const asArray = (v: EventResultRequirements | EventResultRequirement) => (Array.isArray(v) ? v : [v])

function findFirstMissing(requirements: EventResultRequirementsByDate | undefined, results: QualifyingResult[]) {
  if (!requirements || typeof requirements.rules === 'function') {
    return []
  }
  for (const rule of requirements.rules) {
    for (const opt of asArray(rule)) {
      const { count, ...rest } = opt
      if (results.filter((r) => objectContains(r, rest)).length < count) {
        return rest
      }
    }
  }
}

export function availableTypes(requirements?: EventResultRequirementsByDate, eventType?: string) {
  if (!requirements || typeof requirements.rules === 'function') {
    switch (eventType) {
      case 'NOME-B SM':
        return ['NOME-B']
      case 'NOME-A SM':
        return ['NOME-A', 'NOME-A KV']
      case 'NOWT SM':
        return ['NOWT']
    }
    return []
  }
  return unique(requirements.rules.flatMap((rule) => asArray(rule).map((opt) => opt.type)))
}

const availableSMResults = (type?: string, eventType?: string, hasKVA?: boolean) => {
  switch (eventType) {
    case 'NOME-B SM':
      return [hasKVA ? undefined : 'FI KVA-B', 'VOI1', 'VOI2', 'VOI3'].filter(Boolean)
    case 'NOWT SM':
      return [hasKVA ? undefined : 'FI KVA-WT', 'VOI1', 'VOI2', 'VOI3'].filter(Boolean)
    case 'NOME-A SM':
      if (type === 'NOME-A')
        return [hasKVA ? undefined : 'FI KVA-FT', 'A1 CERT', 'A1 RES-CERT', 'A1', 'A2', 'A3'].filter(Boolean)

      if (type === 'NOME-A KV') return ['EXC CACIT', 'EXC RES-CACIT', 'EXC', 'VG', 'G']
  }
  return []
}

export function availableResults(
  requirements?: EventResultRequirementsByDate,
  type?: string,
  eventType?: string,
  existingResults?: ManualTestResult[]
) {
  if (!requirements || typeof requirements.rules === 'function') {
    if (eventType?.includes(' SM')) {
      const hasKVA = existingResults?.some((r) => r.result.startsWith('FI KVA'))

      return availableSMResults(type, eventType, hasKVA)
    }
    return []
  }
  return unique(
    requirements.rules.flatMap((rule) =>
      asArray(rule)
        .filter((opt) => !type || opt.type === type)
        .map((opt) => (opt.cert ? 'CERT' : opt.result))
    )
  )
}

export function createMissingResult(
  requirements: EventResultRequirementsByDate | undefined,
  results: Array<QualifyingResult | ManualTestResult>,
  regNo: string
): ManualTestResult {
  const rule = findFirstMissing(requirements, results)
  return {
    id: nanoid(10),
    regNo,
    date: new Date(),
    official: false,
    qualifying: true,
    type: '',
    judge: '',
    location: '',
    result: '',
    class: '',
    ...rule,
  }
}

export function resultBorderColor(qualifying: boolean | undefined) {
  if (qualifying === true) {
    return 'success.light'
  }
  if (qualifying === false) {
    return 'error.main'
  }
}

export function getResultId(result: ManualTestResult | QualifyingResult) {
  if ('id' in result) {
    return result.id
  }
  return result.type + result.class + result.date.toISOString()
}
