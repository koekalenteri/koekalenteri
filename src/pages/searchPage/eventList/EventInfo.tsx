import type { Grid2Props } from '@mui/material'
import type { ReactNode } from 'react'
import type { PublicDogEvent, PublicJudge } from '../../../types'

import { useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import Box from '@mui/material/Box'
import Typography from '@mui/material/Typography'
import Grid from '@mui/material/Unstable_Grid2/Grid2'
import { endOfDay } from 'date-fns'

import useEventStatus from '../../../hooks/useEventStatus'
import { judgeName } from '../../../lib/judge'
import { isEntryOpen, printContactInfo, unique } from '../../../lib/utils'
import CostInfo from '../../components/CostInfo'
import { PriorityChips } from '../../components/PriorityChips'

import { EventClassInfo } from './eventInfo/EventClassInfo'

interface Props {
  readonly event: PublicDogEvent
}

const InfoItem = ({ label, children, ...props }: { label: string; children: ReactNode } & Grid2Props) => (
  <Grid xs={12} md={6} lg={4} xl={2} {...props}>
    <Typography
      variant="caption"
      color="text.secondary"
      bgcolor="background.form"
      borderRadius="4px"
      sx={{ p: 0.5, ml: -0.5, width: '100%', display: 'block' }}
    >
      {label}
    </Typography>
    {children}
  </Grid>
)

const judgeClasses = (judge: PublicJudge, event: PublicDogEvent) => {
  const classes = unique(
    event.classes
      .filter((c) => (Array.isArray(c.judge) ? c.judge.find((j) => j.id === judge.id) : c.judge?.id === judge.id))
      .map((c) => c.class)
  )
  return classes.length ? ` (${classes.join(', ')})` : ''
}

export const EventInfo = ({ event }: Props) => {
  const { t } = useTranslation()
  const status = useEventStatus(event)
  const official = useMemo(() => printContactInfo(event.contactInfo?.official), [event.contactInfo?.official])
  const secretary = useMemo(() => printContactInfo(event.contactInfo?.secretary), [event.contactInfo?.secretary])
  const classes = useMemo(
    () => (event.classes.length ? unique(event.classes.map((c) => c.class)) : [event.eventType]),
    [event.classes, event.eventType]
  )
  const judges = useMemo(() => event.judges.map((j) => `${judgeName(j, t)}${judgeClasses(j, event)}`), [event, t])

  return (
    <Grid container columnSpacing={1} disableEqualOverflow sx={{ py: 0.5 }}>
      <InfoItem label={t('entryTime')} order={{ xs: 1 }}>
        {t('dateFormat.datespan', { start: event.entryStartDate, end: event.entryEndDate })}{' '}
        <span className="info">{status}</span>
        {isEntryOpen(event) ? t('dateFormat.distanceLeft', { date: endOfDay(event.entryEndDate!) }) : ''}
      </InfoItem>
      {classes.length ? (
        <InfoItem label={t('event.classPlaces')} order={{ xs: 2, lg: 3, xl: 10 }} xl={true}>
          {classes.map((c) => (
            <EventClassInfo key={c} event={event} eventClass={c} />
          ))}
        </InfoItem>
      ) : null}
      <InfoItem label={t('event.judges')} order={{ xs: 3, lg: 2 }}>
        {judges.map((j) => (
          <Box>{j}</Box>
        ))}
      </InfoItem>
      {official ? (
        <InfoItem label={t('event.official')} order={{ xs: 4, lg: 2 }}>
          {official}
        </InfoItem>
      ) : null}
      {secretary ? (
        <InfoItem label={t('event.secretary')} order={{ xs: 5, lg: official ? undefined : 2 }}>
          {secretary}
        </InfoItem>
      ) : null}
      {event.cost ? (
        <InfoItem label={t('paymentDetails')} order={{ xs: 6, xl: 9 }}>
          <CostInfo event={event} />
        </InfoItem>
      ) : null}
      {event.priority?.length ? (
        <InfoItem label={t('event.priority')} order={{ xs: 7 }}>
          <PriorityChips priority={event.priority} />
        </InfoItem>
      ) : null}
      {event.description ? (
        <InfoItem label={t('event.description')} order={{ xs: 8 }}>
          {event.description}
        </InfoItem>
      ) : null}
    </Grid>
  )
}
