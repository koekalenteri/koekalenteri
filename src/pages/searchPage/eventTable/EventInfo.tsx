import type { PublicDogEvent } from '../../../types'

import { useCallback, useMemo } from 'react'
import { useTranslation } from 'react-i18next'
import Table from '@mui/material/Table'
import TableBody from '@mui/material/TableBody'
import TableCell from '@mui/material/TableCell'
import TableRow from '@mui/material/TableRow'

import useEventStatus from '../../../hooks/useEventStatus'
import { isEntryOpen, printContactInfo } from '../../../lib/utils'
import CostInfo from '../../components/CostInfo'
import { PriorityChips } from '../../components/PriorityChips'
import { useJudgesActions } from '../../recoil'

import { EventClassRow } from './eventInfo/EventClassRow'

interface Props {
  readonly event: PublicDogEvent
}

export const EventInfo = ({ event }: Props) => {
  const { t } = useTranslation()
  const judgeActions = useJudgesActions()
  const status = useEventStatus(event)

  const judgeName = useCallback((id: number) => judgeActions.find(id)?.name ?? '', [judgeActions])
  const haveJudgesWithoutAssignedClass = useMemo(
    () =>
      event.judges.filter(
        (j) =>
          !event.classes.find((c) => (Array.isArray(c.judge) ? c.judge.find((cj) => cj.id === j) : c.judge?.id === j))
      ).length > 0,
    [event.classes, event.judges]
  )

  return (
    <Table
      size="small"
      aria-label="details"
      sx={{
        border: 0,
        my: 0.5,
        mx: 0,
        '& *': {
          border: 0,
          my: 0,
          borderSpacing: 0,
          padding: '2px 16px 2px 0',
        },
        '& th': {
          border: 0,
          margin: 0,
          py: 0,
          pl: 0,
          borderSpacing: 0,
          width: '1%',
          whiteSpace: 'nowrap',
          verticalAlign: 'top',
        },
        '& td': {
          border: 0,
          margin: 0,
          py: 0,
          pl: 0,
          borderSpacing: 0,
          whiteSpace: 'normal',
        },
      }}
    >
      <TableBody>
        <TableRow key={event.id + 'date'}>
          <TableCell component="th" scope="row">
            {t('entryTime')}:
          </TableCell>
          <TableCell>
            {t('daterange', { start: event.entryStartDate, end: event.entryEndDate })}
            <span className="info">{status}</span>
            {isEntryOpen(event) ? t('dateFormat.distanceLeft', { date: event.entryEndDate }) : ''}
          </TableCell>
        </TableRow>
        {event.classes.length !== 0 && <EventClassRow key={event.id + 'classes'} event={event} />}
        {haveJudgesWithoutAssignedClass && (
          <>
            <TableRow key={event.id + 'judge' + event.judges[0]}>
              <TableCell component="th" scope="row" rowSpan={event.judges.length}>
                {t('event.judges')}:
              </TableCell>
              <TableCell>{judgeName(event.judges[0])}</TableCell>
            </TableRow>
            {event.judges.slice(1).map((judgeId) => (
              <TableRow key={event.id + 'judge' + judgeId}>
                <TableCell>{judgeName(judgeId)}</TableCell>
              </TableRow>
            ))}
          </>
        )}
        {event.contactInfo?.official ? (
          <TableRow key={event.id + 'official'}>
            <TableCell component="th" scope="row">
              {t('event.official')}:
            </TableCell>
            <TableCell>{printContactInfo(event.contactInfo?.official)}</TableCell>
          </TableRow>
        ) : null}
        {event.contactInfo?.secretary ? (
          <TableRow key={event.id + 'secretary'}>
            <TableCell component="th" scope="row">
              {t('event.secretary')}:
            </TableCell>
            <TableCell>{printContactInfo(event.contactInfo?.secretary)}</TableCell>
          </TableRow>
        ) : null}
        <TableRow key={event.id + 'payment'}>
          <TableCell component="th" scope="row">
            {t('paymentDetails')}:
          </TableCell>
          <TableCell>
            <CostInfo event={event} />
          </TableCell>
        </TableRow>
        {event.priority?.length ? (
          <TableRow key={event.id + 'priority'}>
            <TableCell component="th" scope="row">
              {t('event.priority')}:
            </TableCell>
            <TableCell>
              <PriorityChips priority={event.priority} />
            </TableCell>
          </TableRow>
        ) : null}
        {event.description ? (
          <TableRow key={event.id + 'description'}>
            <TableCell component="th" scope="row">
              {t('event.description')}:
            </TableCell>
            <TableCell>{event.description}</TableCell>
          </TableRow>
        ) : null}
      </TableBody>
    </Table>
  )
}
