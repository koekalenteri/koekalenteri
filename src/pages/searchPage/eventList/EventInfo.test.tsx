import type { DogEvent } from '../../../types'

import { Suspense } from 'react'
import { render, screen } from '@testing-library/react'
import { RecoilRoot } from 'recoil'

import { emptyEvent } from '../../../__mockData__/emptyEvent'
import { flushPromises } from '../../../test-utils/utils'

import { EventInfo } from './EventInfo'

jest.mock('../../../api/judge')

const testEvent: DogEvent = {
  ...emptyEvent,
  organizer: {
    id: '0',
    name: 'test organization',
  },
  name: 'name',
  location: 'location',
  startDate: new Date('2021-02-10'),
  endDate: new Date('2021-02-11'),
  entryStartDate: new Date('2021-01-20'),
  entryEndDate: new Date('2021-02-04'),
  description: 'event description text',
  classes: [
    {
      date: new Date('2021-02-10'),
      class: 'ALO',
      judge: { id: 1, name: 'Test Judge' },
      places: 11,
      entries: 22,
      members: 2,
    },
  ],
  places: 11,
  entries: 22,
  members: 2,
}

describe('EventInfo', () => {
  beforeAll(() => jest.useFakeTimers())
  afterEach(() => jest.runOnlyPendingTimers())
  afterAll(() => jest.useRealTimers())

  it('should render event information', async () => {
    const event: DogEvent = { ...testEvent }
    const { container } = render(
      <RecoilRoot>
        <Suspense fallback={<div>loading...</div>}>
          <EventInfo event={event} />
        </Suspense>
      </RecoilRoot>
    )
    await flushPromises()

    expect(container).toMatchSnapshot()
  })

  it('should render contact info', async () => {
    const event: DogEvent = {
      ...testEvent,
      contactInfo: {
        official: { name: 'official name', email: 'official@example.com', phone: '0700-official' },
        secretary: { name: 'secretary name', email: 'secretary@example.com', phone: '0700-secretary' },
      },
    }
    const { container } = render(
      <RecoilRoot>
        <Suspense fallback={<div>loading...</div>}>
          <EventInfo event={event} />
        </Suspense>
      </RecoilRoot>
    )
    await flushPromises()

    expect(screen.queryByText('event.official')).toBeInTheDocument()
    expect(screen.queryByText('official name, 0700-official, official@example.com')).toBeInTheDocument()
    expect(screen.queryByText('event.secretary')).toBeInTheDocument()
    expect(screen.queryByText('secretary name, 0700-secretary, secretary@example.com')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('should not render empty contact info', async () => {
    const event: DogEvent = {
      ...testEvent,
      contactInfo: {
        official: {},
        secretary: {},
      },
    }
    const { container } = render(
      <RecoilRoot>
        <Suspense fallback={<div>loading...</div>}>
          <EventInfo event={event} />
        </Suspense>
      </RecoilRoot>
    )
    await flushPromises()

    expect(screen.queryByText('event.official:')).toBeNull()
    expect(screen.queryByText('event.secretary:')).toBeNull()
  })
})
