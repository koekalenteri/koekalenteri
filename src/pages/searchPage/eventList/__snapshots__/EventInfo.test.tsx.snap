// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventInfo should render contact info 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-b1kjk3-MuiGrid2-root"
  >
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-yzd260-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        entryTime
      </span>
      dateFormat.datespan
       
      <span
        class="info"
      />
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-true css-yu5d00-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.classPlaces
      </span>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1hltvp8-MuiTypography-root"
          >
            event.classPlacesHeader.name
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1hltvp8-MuiTypography-root"
          >
            event.classPlacesHeader.dates
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-tx3obz-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1hltvp8-MuiTypography-root"
          >
            event.classPlacesHeader.entries
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-tx3obz-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1hltvp8-MuiTypography-root"
          >
            event.classPlacesHeader.places
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-tx3obz-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1hltvp8-MuiTypography-root"
          >
            event.classPlacesHeader.members
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-dr8jig-MuiGrid-root"
        >
          ALO
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1u5voze-MuiGrid-root"
        >
          eeeeee d.M.
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-83otcn-MuiGrid-root"
        >
          0
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-83otcn-MuiGrid-root"
        >
          10
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-83otcn-MuiGrid-root"
        >
          2
        </div>
      </div>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-241rwc-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.judges
      </span>
      <div
        class="MuiBox-root css-0"
      >
        Tuomari 1
      </div>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-1dgrg5x-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.official
      </span>
      official name, 0700-official, official@example.com
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-m9ba3e-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.secretary
      </span>
      secretary name, 0700-secretary, secretary@example.com
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-1i1pohp-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        paymentDetails
      </span>
      123 €event.costMember 123 €
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-1e29lse-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.priority
      </span>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-13htn8f-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          Jäsenet
        </span>
      </div>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-kjsw1z-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.description
      </span>
      event description text
    </div>
  </div>
</div>
`;

exports[`EventInfo should render event information 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-b1kjk3-MuiGrid2-root"
  >
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-yzd260-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        entryTime
      </span>
      dateFormat.datespan
       
      <span
        class="info"
      />
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-true css-yu5d00-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.classPlaces
      </span>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1hltvp8-MuiTypography-root"
          >
            event.classPlacesHeader.name
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1hltvp8-MuiTypography-root"
          >
            event.classPlacesHeader.dates
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-tx3obz-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1hltvp8-MuiTypography-root"
          >
            event.classPlacesHeader.entries
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-tx3obz-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1hltvp8-MuiTypography-root"
          >
            event.classPlacesHeader.places
          </span>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-tx3obz-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1hltvp8-MuiTypography-root"
          >
            event.classPlacesHeader.members
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-dr8jig-MuiGrid-root"
        >
          ALO
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-1u5voze-MuiGrid-root"
        >
          eeeeee d.M.
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-83otcn-MuiGrid-root"
        >
          0
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-83otcn-MuiGrid-root"
        >
          10
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-83otcn-MuiGrid-root"
        >
          2
        </div>
      </div>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-241rwc-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.judges
      </span>
      <div
        class="MuiBox-root css-0"
      >
        Tuomari 1
      </div>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-1dgrg5x-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.official
      </span>
      Teemu Toimitsija, 040-official, official@example.com
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-m9ba3e-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.secretary
      </span>
      Siiri Sihteeri, 040-secretary, secretary@example.com
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-1i1pohp-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        paymentDetails
      </span>
      123 €event.costMember 123 €
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-1e29lse-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.priority
      </span>
      <div
        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-13htn8f-MuiChip-root"
      >
        <span
          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
        >
          Jäsenet
        </span>
      </div>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 MuiGrid2-grid-lg-4 MuiGrid2-grid-xl-2 css-kjsw1z-MuiGrid2-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-ltaje5-MuiTypography-root"
      >
        event.description
      </span>
      event description text
    </div>
  </div>
</div>
`;
